perftest/wallet.go

```diff
@@ -0,0 +1,425 @@
+package perftest
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strings"
+	"time"
+
+	proto "code.vegaprotocol.io/protos/vega"
+)
+
+// UserDetails Holds wallet information for each user
+type UserDetails struct {
+	userName string
+	token    string
+	pubKey   string
+}
+
+func abs(value int64) int64 {
+	if value < 0 {
+		return -value
+	}
+	return value
+}
+
+// SecondsFromNowInSecs : Creates a timestamp relative to the current time in seconds
+func SecondsFromNowInSecs(seconds int64) int64 {
+	return time.Now().Unix() + seconds
+}
+
+func loginWallet(walletURL, userName, password string) (string, error) {
+	postBody, _ := json.Marshal(map[string]string{
+		"wallet":     userName,
+		"passphrase": password,
+	})
+	responseBody := bytes.NewBuffer(postBody)
+
+	URL := "http://" + walletURL + "/api/v1/auth/token"
+
+	resp, err := http.Post(URL, "application/json", responseBody)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+	sb := string(body)
+	if strings.Contains(sb, "error") {
+		return "", fmt.Errorf(sb)
+	}
+
+	// Get the token value out from the JSON
+	var result map[string]interface{}
+	json.Unmarshal([]byte(sb), &result)
+
+	return result["token"].(string), nil
```

Comment:
just to be on the safe side, we could do `resStr, ok := result["token"].(string)` and then `if !ok { return err... }`.
otherwise, if "token" is not in the map, we get a panic